diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/PsiUtils.kt

```diff
@@ -0,0 +1,94 @@
+                    (selectorExpression is KtReferenceExpression ||
+                            ((selectorExpression as? KtCallExpression)
+                                    ?.valueArgumentList
+                                    ?.arguments
+                                    ?.all { it.getArgumentExpression()!!.containsOnlyConstants() }
+                                    ?: false)
+                            )
+            else -> false
+        }
+
+/**
+ * Get block inside which the property is declared.
+ * Here we assume that property can be declared only in block, since declaration is not an expression in kotlin
+ * and compiler prohibits things like `if (condition) val x = 0`.
+ */
+fun KtProperty.getDeclarationScope() = getParentOfType<KtBlockExpression>(true)!!
+        .let { if (it is KtIfExpression) it.then!! else it }
+        .let { if (it is KtTryExpression) it.tryBlock else it }
+        as KtBlockExpression
+
+/**
+ * Finds all references to [property] in the same code block.
+ * @return list of references as [KtNameReferenceExpression]
+ */
+fun findUsagesOf(property: KtProperty) = property
+        .getDeclarationScope()
+        .let { declarationScope ->
+            val name = property.nameAsName
+            declarationScope
+                    .node
+                    .findAllNodesWithSpecificType(ElementType.REFERENCE_EXPRESSION)
+                    .map { it.psi as KtNameReferenceExpression }
+                    .filter { it.getReferencedNameAsName() == name }
+                    .filterNot {
+                        // to avoid false triggering on objects' fields with same name as local property
+                        (it.parent as? KtDotQualifiedExpression)?.run {
+                            receiverExpression != it && selectorExpression?.referenceExpression() == it
+                        } ?: false
+                    }
+                    .filterNot { ref ->
+                        // to exclude usages of local properties and lambda arguments with same name
+                        ref.parents.mapNotNull { it as? KtBlockExpression }.takeWhile { it != declarationScope }.any { block ->
+                            block.getChildrenOfType<KtProperty>().any { it.nameAsName == name } ||
+                                    (block.parent.let { it as? KtFunctionLiteral }?.valueParameters?.any { it.nameAsName == name }
+                                            ?: false)
+                        }
+                    }
+        }
+
+/**
+ * Checks if this [PsiElement] is an ancestor of [block].
+ * Nodes like `IF`, `TRY` are parents of `ELSE`, `CATCH`, but their scopes are not intersecting, and false is returned in this case.
+ */
+fun PsiElement.isContainingScope(block: KtBlockExpression): Boolean {
+    when (block.parent.node.elementType) {
+        ElementType.ELSE -> getParentOfType<KtIfExpression>(true)
+        ElementType.CATCH -> getParentOfType<KtTryExpression>(true)
```

Comment:
our code style recommends to make else -> branch

Reply:
Yes, it has disappeared somehow, thx for noticing